scanpipe/pipes/windows.py

```diff
@@ -0,0 +1,250 @@
+    lookup = Q(rootfs_path__startswith=root_dir_pattern)
+
+    # If there are Q() objects in `q_objects`, then those Q() objects are chained
+    # to the initial query `lookup` using AND to allow a more specific query for
+    # package files.
+    for q_object in q_objects:
+        lookup &= q_object
+
+    installed_package_files = qs.filter(lookup)
+    # If we find files whose names start with `root_dir_pattern`, we consider
+    # these files to be part of the Package `package` and tag these files as such.
+    if installed_package_files:
+        created_package = pipes.update_or_create_package(project, package.to_dict())
+        for installed_package_file in installed_package_files:
+            installed_package_file.discovered_packages.add(created_package)
+            installed_package_file.status = "installed-package"
+            installed_package_file.save()
+        created_package.save()
+
+
+def _tag_python_software(project):
+    qs = project.codebaseresources.no_status()
+    python_root_pattern = r"(^/(Files/)?Python(\d+)?)/.*$"
+    python_root_pattern_compiled = re.compile(python_root_pattern)
+
+    python_versions_by_path = {}
+    for python_resource in qs.filter(rootfs_path__regex=python_root_pattern):
+        _, python_root_dir, _, version, _ = re.split(
+            python_root_pattern_compiled,
+            python_resource.rootfs_path,
+        )
+        if python_root_dir in python_versions_by_path:
+            continue
+        if not version:
+            version = "nv"
+        if version != "nv":
+            version = ".".join(digit for digit in version)
+        python_versions_by_path[python_root_dir] = version
+
+    # We do not want to tag the files in the `site-packages` directory as being
+    # from Python proper. The packages found here are oftentime third-party
+    # packages from outside the Python foundation
+    q_objects = [~Q(rootfs_path__icontains="site-packages")]
+    for python_path, python_version in python_versions_by_path.items():
+        python_package = win_reg.InstalledWindowsProgram(
+            name="Python",
+            version=python_version,
+            license_expression="python",
+            copyright="Copyright (c) Python Software Foundation",
+            homepage_url="https://www.python.org/",
+        )
+        tag_installed_package_files(
+            project=project,
+            root_dir_pattern=python_path,
+            package=python_package,
+            q_objects=q_objects,
+        )
+
+
+def _tag_openjdk_software(project):
+    qs = project.codebaseresources.no_status()
+    openjdk_root_pattern = r"^(/(Files/)?(open)?jdk(-((\d*)(\.\d+)*))*)/.*$"
+    openjdk_root_pattern_compiled = re.compile(openjdk_root_pattern)
+
+    openjdk_versions_by_path = {}
+    for openjdk_codebase_resource in qs.filter(rootfs_path__regex=openjdk_root_pattern):
+        _, openjdk_root_path, _, _, _, openjdk_version, _, _, _ = re.split(
```

Comment:
What about using the match system https://docs.python.org/3/library/re.html#match-objects for improved readability and future maintenance?
Some examples in https://github.com/package-url/packageurl-python/blob/master/src/packageurl/contrib/url2purl.py

Reply:
This applies to the 3 `re.split()` of the PR.

Reply:
I replaced all useage of `re.split()` with `re.match()`